### PR TITLE
Minimal dependencyManagement, jboss-logging, gmaven

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -133,6 +133,8 @@
     <version.maven-source-plugin>2.4</version.maven-source-plugin>
     <version.maven-surefire-plugin>2.17</version.maven-surefire-plugin>
     <version.maven-war-plugin>2.5</version.maven-war-plugin>
+    <version.org.codehaus.gmavenplus.gmavenplus-plugin>1.2</version.org.codehaus.gmavenplus.gmavenplus-plugin>
+    <version.org.bsc.maven.maven-processor-plugin>2.0.2</version.org.bsc.maven.maven-processor-plugin>
```
Comment (jmazzitelli) [inline pom.xml:136]: I have  a very trivial comment to make - if we want this list in alphabetical order, "org.codehaus..." should come after "org.bsc..." - so switch lines 136 and 137.

**Comment by jmazzitelli** [comment]
I tested my stuff with this PR and it works fine. I'm +1 on merging this.

**Comment by jmazzitelli** [comment]
Note that I think maven.compiler.source is not configured fully in the parent pom.xml.

We have these defined:

```
<!-- maven-compiler-plugin -->
<maven.compiler.source>1.8</maven.compiler.source>
<maven.compiler.target>1.8</maven.compiler.target>
<maven.compiler.showWarnings>true</maven.compiler.showWarnings>
```

however, they are not used in the plugin config:

```
  <plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-compiler-plugin</artifactId>
    <configuration>
      <source>${maven.compiler.source}</source> <----- this is missing
      <target>${maven.compiler.target}</target> <----- this is missing
      <showWarnings>${maven.compiler.showWarnings}</showWarnings> <--- this wasn't using the property
      <optimize>true</optimize>
      <debug>true</debug>
      <showDeprecation>true</showDeprecation>
    </configuration>
  </plugin>
```

I believe this is why my Eclipse keeps updating my projects such that they are requiring Java 1.4 compliance.

File: pom.xml (hunk 1 above)
Comment (ppalaga) [inline pom.xml:136]: +1 thanks for finding it :)

**Comment by ppalaga** [comment]
No, explicit `<source>${maven.compiler.source}</source> <target>${maven.compiler.target}</target>` in plugin's config is not necessary, as those two property names are hard coded in the plugin to be picked up as "user props" http://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#source

**Pull request merged**